kernel/ptrace.c

```diff
@@ -0,0 +1,243 @@
+#include "ptrace.h"
+#include "kernel/calls.h"
+#include "kernel/errno.h"
+#include "kernel/signal.h"
+#include "task.h"
+#include <string.h>
+
+// Returns stopped child with the given pid, locked with stop_lock
+static struct task *find_child(struct task *parent, pid_t_ pid) {
```

Comment:
Is it necessary to have the task parameter if it's always `current`?

Reply:
Well, there's multiple problems with it. One is that yet, the parameter is useless (I'll remove it), and another is that you can trace things that aren't your children. But since we don't have attach yet this function "works" for now.